File: android-sdk/src/main/java/com/checkout/android_sdk/Models/CardModel.java
package com.checkout.android_sdk.Models;

/**
 * Http request card details object model
 */
public class CardModel {

    private String expiryMonth;
    private String expiryYear;
    private BillingModel billingDetails;
    private String id;
    private String last4;
    private String bin;
    private String paymentMethod;
    private String name;

    public String getExpiryMonth() {
        return expiryMonth;
    }

    public String getExpiryYear() {
        return expiryYear;
    }

    public BillingModel getBillingDetails() {
        return billingDetails;
    }

    public String getId() {
        return id;
    }

    public String getLast4() {
        return last4;
    }

    public String getBin() {
        return bin;
    }

    public String getPaymentMethod() {
        return paymentMethod;
    }

    public String getName() {
        return name;
    }
}

File: android-sdk/src/main/java/com/checkout/android_sdk/Models/CardModel.kt
package com.checkout.android_sdk.Models

import com.google.gson.annotations.SerializedName

/**
 * Http request card details object model
 */
class CardModel {
    @SerializedName("expiryMonth")
    val expiryMonth: String? = null
    @SerializedName("expiryYear")
    val expiryYear: String? = null
    @SerializedName("billingDetails")
    val billingDetails: BillingModel? = null
    @SerializedName("id")
    val id: String? = null
    @SerializedName("last4")
    val last4: String? = null
    @SerializedName("bin")
    val bin: String? = null
    @SerializedName("paymentMethod")
    val paymentMethod: String? = null
    @SerializedName("name")
    val name: String? = null
}
